lib/streamlit/state/session_state.py

```diff
@@ -114,21 +263,73 @@ def __getattr__(self, key: str) -> Any:
     def __setattr__(self, key: str, value: Any) -> None:
         # Setting the _old_state and _new_state attributes must be done using
         # the base method to avoid recursion.
-        if key in ["_new_state", "_old_state"]:
+        if key in ["_new_session_state", "_new_widget_state", "_old_state"]:
             super().__setattr__(key, value)
         else:
             self[key] = value
 
     def __delattr__(self, key: str) -> None:
+        if key in ["_new_session_state", "_new_widget_state", "_old_state"]:
```

Comment:
Should this be `not in`?

Also, I think once `__delitem__` is reworked, this won't be necessary since we'll just be able to catch and re-raise the `KeyError` thrown from that method

Reply:
Whoops nevermind. I'm not sure how I managed to not realize the point of this line initially